gui/network/forms.py

```diff
@@ -588,153 +594,29 @@ def __init__(self, *args, **kwargs):
-                        "filling out nameserver 3")
-                self._errors["gc_nameserver3"] = self.error_class([msg])
-                msg = _("Required when using nameserver 3")
-                self._errors["gc_nameserver1"] = self.error_class([msg])
-                del cleaned_data["gc_nameserver1"]
-            if nameserver1 == "" or nameserver2 == "":
-                del cleaned_data["gc_nameserver3"]
-        elif nameserver2:
-            if nameserver1 == "":
-                del cleaned_data["gc_nameserver2"]
-                msg = _("Must fill out nameserver 1 before "
-                        "filling out nameserver 2")
-                self._errors["gc_nameserver2"] = self.error_class([msg])
-                msg = _("Required when using nameserver 3")
-                self._errors["gc_nameserver1"] = self.error_class([msg])
-                del cleaned_data["gc_nameserver1"]
-        return cleaned_data
+        return str(self.cleaned_data['gc_nameserver3'])
 
-    def save(self):
-        # TODO: new IP address should be added in a side-by-side manner
-        # or the interface wouldn't appear once IP was changed.
-        retval = super(GlobalConfigurationForm, self).save()
-
-        whattoreload = ["hostname"]
-        if (
-            self.instance._orig_gc_domain != self.cleaned_data.get('gc_domain') or
-            self.instance._orig_gc_nameserver1 != self.cleaned_data.get('gc_nameserver1') or
-            self.instance._orig_gc_nameserver2 != self.cleaned_data.get('gc_nameserver2') or
-            self.instance._orig_gc_nameserver3 != self.cleaned_data.get('gc_nameserver3')
-        ):
-            # Note notifier's _reload_resolvconf has reloading hostname folded in it
-            whattoreload.append("resolvconf")
-        if (
-            self.instance._orig_gc_ipv4gateway != self.cleaned_data.get('gc_ipv4gateway') or
-            self.instance._orig_gc_ipv6gateway != self.cleaned_data.get('gc_ipv6gateway')
-        ):
-            # this supersedes all since it has hostname and resolvconf reloads folded in it
-            whattoreload.append("networkgeneral")
-            with client as c:
-                c.call('routes.sync')
-
-        if hasattr(notifier, 'failover_licensed') and notifier().failover_licensed() and \
-                self.instance._orig_gc_hostname_virtual != self.cleaned_data.get('gc_hostname_virtual'):
-            from freenasUI.services.models import services, NFS
-            svcobj = services.objects.get(srv_service='nfs')
-            nfsobj = NFS.objects.all()[0]
-            if (svcobj and svcobj.srv_enable) and (nfsobj and (nfsobj.nfs_srv_v4 and nfsobj.nfs_srv_v4_krb)):
-                notifier().restart("ix-nfsd")
-                whattoreload.append("mountd")
-
-        for what in whattoreload:
-            notifier().reload(what)
+    def middleware_clean(self, update):
+        update['domains'] = update['domains'].split()
```

Comment:
`list(filter(None, re.split(r"\s+", update["domains"])))` here and everywhere else would work better